Send content_id to content store and panopticon

tijmenb opened the pull request:
All taggable content on GOV.UK needs a content_id. This PR adds the content_id to the data being sent to panopticon and adds a rake task to send the item to the publishing-api.

Trello: https://trello.com/c/ZqWkgsej

tijmenb committed:
Extract pge data into class

Panopticon and content-store publishers will use this data.

tijmenb committed:
Add PageContentItem presenter

This class generates a schema-valid content-store item.

tijmenb committed:
Add content_id to page data

By including this here, the content_id will be automatically sent to
panopticon. The content_id was generated by running `SecureRandom.uuid`
in the rails console.

tijmenb committed:
Send content_id to the content-store

This will allow us to tag the page.

tijmenb committed:
Add description to content-item

Also uses the title from the data object for consistency sake.

tijmenb committed:
Add rake task to send content to publishing-api

We want to send this item to the content-store so it can be tagged.

tijmenb committed:
Update jenkins.sh

This downloads the govuk-content-schemas so the schema tests pass.

Script copied from:
https://raw.githubusercontent.com/alphagov/calculators/master/jenkins.sh

bishboria commented:
:+1:

bishboria commented:
(Leaving the PR open for a bit longer, in case others want to review)

mobaig commented on spec/notifiers/publishing_api_notifier_spec.rb:
```diff
@@ -0,0 +1,16 @@
+
+      expect(Services.publishing_api).to have_received(:put_content_item).with(
+        "/business-finance-support-finder",
+        be_valid_against_schema('placeholder')
```
As discussed, this could be re-worked to be clearer.

tijmenb committed:
Refactor publishing-api assertion

The current assertion is difficult to read. If we rework this to
block-style it's clearer that we are using the
`be_valid_against_schema` on the second argument to
`#put_content_item`.

KushalP commented:
Are all reviewers done (and happy) with this PR?

mobaig commented:
LGTM
